pubsub/tests/unit/subscriber_test.py

```diff
@@ -452,6 +477,46 @@ async def f(*args):
         assert nack_queue.qsize() == 1
         assert queue.qsize() == 0
 
+    @pytest.mark.asyncio
+    async def test_consumer_gracefull_shutdown(
+        ack_deadline_cache, message
+    ):
+        queue = asyncio.Queue()
+        ack_queue = asyncio.Queue()
+        nack_queue = asyncio.Queue()
+        mock = MagicMock()
+        event = asyncio.Event()
+
+        async def f(*args):
+            mock(*args)
+            await event.wait()
+
+        consumer_task = asyncio.ensure_future(
+            consumer(
+                queue,
+                f,
+                ack_queue,
+                ack_deadline_cache,
+                1,
+                nack_queue,
+                MagicMock()
+            )
+        )
+        await queue.put((message, 0.0))
+        await asyncio.sleep(0)
+        await asyncio.sleep(0)
+        mock.assert_called_once()
+        consumer_task.cancel()
+        await asyncio.sleep(0)
+        assert not consumer_task.done()
+        event.set()
+        await asyncio.sleep(0)
+        assert ack_queue.qsize() == 1
+        await ack_queue.get()
+        ack_queue.task_done()
+        await asyncio.sleep(0.3)
```

Comment:
Maybe a comment of why we need 0.3 here?

Reply:
Even better, we don't need `0.3` here anymore!